src/protocol/frame/frame.rs

```diff
@@ -161,14 +160,18 @@ impl FrameHeader {
             let length_byte = second & 0x7F;
             let length_length = LengthFormat::for_byte(length_byte).extra_bytes();
             if length_length > 0 {
-                match cursor.read_uint::<NetworkEndian>(length_length) {
-                    Err(ref err) if err.kind() == ErrorKind::UnexpectedEof => {
-                        return Ok(None);
+                const SIZE: usize = mem::size_of::<u64>();
+                let mut buffer = [0; SIZE];
+                let start = match SIZE.checked_sub(length_length) {
+                    Some(start) => start,
+                    None => {
+                        panic!("the integer can fit {} bytes, but {} is given", SIZE, length_length)
```

Comment:
Nit pick.

Wouldn't it be better to use the `unreachable!` macros here?

It is the same as `panic!` in terms of generated code, but better clarify that `websocket` length max is 8 bytes, exactly `sizeof::<u64>`, so it is impossible condition.

And even with rustc 1.63.0, it is possible to write:

```
               const fn extra_bytes(&self) -> usize {...}
                const SIZE: usize = LengthFormat::U64.extra_bytes();
```

so we can guarantee that `u64::from_be_bytes` can handle max websocket packet length,
and replace `panic!` with ` std::intrinsics::unreachable`, to make code faster.
But I suppose it is overkill to use `unsafe` for such case.

Reply:
An `assert!(length_length <= SIZE, panic_msg);` would let you use normal subtraction and is more direct, I think. Maybe that would express the unreachability enough?